samcli/commands/local/lib/local_function_url_service.py

```diff
@@ -0,0 +1,351 @@
+            self._shutdown_event.set()
+
+        signal.signal(signal.SIGINT, signal_handler)
+        signal.signal(signal.SIGTERM, signal_handler)
+
+        # Start services
+        self.executor = ThreadPoolExecutor(max_workers=len(self.function_urls))
+
+        try:
+            # Start each function service
+            for func_name, func_config in self.function_urls.items():
+                port = self._allocate_port()
+                service = self._start_function_service(func_name, func_config, port)
+                self.services[func_name] = service
+
+                # Start the service (this runs Flask in a thread)
+                service.start()
+
+                # Wait for the service to be ready
+                if not self._wait_for_service(port):
+                    LOG.warning(f"Service for {func_name} on port {port} did not start properly")
+
+            # Print startup info
+            self._print_startup_info()
+
+            # Wait for shutdown signal
+            self._shutdown_event.wait()
+
+        except KeyboardInterrupt:
+            LOG.info("Received keyboard interrupt")
+        finally:
+            self._shutdown_services()
+
+    def start_all(self):
+        """
+        Start all Function URL services. Alias for start() method.
+        """
+        return self.start()
+
+    def start_function(self, function_name: str, port: int):
+        """
+        Start a specific function URL service on the given port.
+
+        Args:
+            function_name: Name of the function to start
+            port: Port to bind the service to
+        """
+        if function_name not in self.function_urls:
+            raise NoFunctionUrlsDefined(f"Function {function_name} does not have a Function URL configured")
+
+        # Setup signal handlers
+        def signal_handler(sig, frame):
+            LOG.info("Received interrupt signal. Shutting down...")
+            self._shutdown_event.set()
+
+        signal.signal(signal.SIGINT, signal_handler)
+        signal.signal(signal.SIGTERM, signal_handler)
+
+        function_url_config = self.function_urls[function_name]
+        service = self._start_function_service(function_name, function_url_config, port)
+        self.services[function_name] = service
+
+        # Start the service (this runs Flask in a thread)
+        service.start()
+
+        # Start service in thread
+        self.executor = ThreadPoolExecutor(max_workers=1)
+
+        # Print startup info for single function
+        url = f"http://{self.host}:{port}/"
+        auth_type = function_url_config["auth_type"]
+        cors_enabled = bool(function_url_config.get("cors"))
+
+        print("\\n" + "=" * 60)
+        print("SAM Local Function URL")
+        print("=" * 60)
+        print(f"\\n  {function_name}:")
+        print(f"    URL: {url}")
+        print(f"    Auth: {auth_type}")
+        print(f"    CORS: {'Enabled' if cors_enabled else 'Disabled'}")
+        print("\\n" + "=" * 60)
```

Comment:
This seems like a different format that the one in `_print_startup_info` . Is there a reason for that? Is that other one better for multiple Furls compared to a single one? We should probably unify